### PR TITLE
VC: selective consistency ns checks 

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

link #1377

### CODE REVIEW

File: incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go [hunk 1]
```diff
@@ -76,98 +79,90 @@ func (c *controller) shouldBeGabageCollected(ns *v1.Namespace) bool {
...
-		for _, clusterName := range clusterNames {
-			wg.Add(1)
-			go func(clusterName string) {
-				defer wg.Done()
-				c.checkNamespacesOfTenantCluster(clusterName)
-			}(clusterName)
-		}
-		wg.Wait()
-	} else {
-		klog.Infof("tenant masters has no clusters, still check pNamespace for gc purpose.")
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go:94]: This is a behavior change, we should continue to check pNamespace for gc purpose when no cluster exists.

File: incubator/virtualcluster/pkg/syncer/util/helper.go [hunk 2]
```diff
@@ -36,3 +40,21 @@ func GetVirtualClusterObject(mc *mc.MultiClusterController, clustername string)
 
 	return vc, nil
 }
+
+func IsObjectScheduleToCluster(obj metav1.Object, clusterID string) error {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/syncer/util/helper.go:44]: I suggest to rename it to "IsNamespaceScheduledToCluster". The logic in the method only applies to Namespace.

File: incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go [hunk 3]
```diff
@@ -76,98 +79,90 @@ func (c *controller) shouldBeGabageCollected(ns *v1.Namespace) bool {
...
+		c.deleteNamespace(p)
+	}
 
-		if err != nil {
-			klog.Errorf("error getting pNamespace %s from super master cache: %v", targetNamespace, err)
-		}
+	vSet.Difference(pSet, differ.FilteringHandler{
+		Handler: d,
+		FilterFunc: func(obj differ.ClusterObject) bool {
+			if obj.OwnerCluster == "" && obj.GetAnnotations()[constants.LabelVCRootNS] == "true" {
```
Comment (Fei-Guo) [inline incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go:152]: Is it ok to remove the obj.OwnerCluster == "" condition?

File: incubator/virtualcluster/pkg/syncer/util/helper.go (hunk 2 above)
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/util/helper.go]: updated

File: incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go (hunk 3 above)
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go:151]: just skip vObj

File: incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go (hunk 1 above)
Comment (zhuangqh) [reply incubator/virtualcluster/pkg/syncer/resources/namespace/checker.go:94]: you are right

**Pull request merged**